src/dmd/dcast.d

```diff
@@ -2177,7 +2177,7 @@ Expression castTo(Expression e, Scope* sc, Type t, Type att = null)
             ArrayLiteralExp ae = e;
 
             Type tb = t.toBasetype();
-            if (tb.ty == Tarray && global.params.vsafe)
+            if (tb.ty == Tarray && global.params.useDIP1000 == FeatureState.enabled)
```

Comment:
The code is way too verbose and repetitive. I suggest using a simple UFCE one-liner:

```suggestion
            if (tb.ty == Tarray && global.params.dip1000.on)
```

```D
module dmd.globals;

// ...

bool on(FeatureState fs) @safe nothrow pure @nogc
{
    return fs == FeatureState.enabled;
}

// ...

extern (C++) struct Param
{
    // ...
    FeatureState dip1000; // implement https://dlang.org/spec/memory-safe-d.html#scope-return-params
    // ...
}
```

Reply:
I'm not sure it's worth it.